src/Common/ThreadStatus.cpp

```diff
@@ -228,6 +226,9 @@ ThreadStatus::~ThreadStatus()
 
     chassert(!check_current_thread_on_destruction || current_thread == this);
 
+    /// Flush untracked_memory **before** switching the current_thread (to avoid losing untracked_memory)
```

Comment:
This comment is quite confusing as is. It should say something like `Flush untracked_memory **after** calling deleter()`, right?